packages/validator/src/services/doppelgangerService.ts

```diff
@@ -0,0 +1,167 @@
+import {Epoch, ValidatorIndex} from "@chainsafe/lodestar-types";
+import {Api} from "@chainsafe/lodestar-api";
+import {ILogger, sleep} from "@chainsafe/lodestar-utils";
+import {IChainForkConfig} from "@chainsafe/lodestar-config";
+import {computeEndSlotForEpoch} from "@chainsafe/lodestar-beacon-state-transition";
+import {AbortController} from "@chainsafe/abort-controller";
+import {PubkeyHex} from "../types";
+import {IClock} from "../util";
+import {IndicesService} from "./indices";
+import {LivenessResponseData} from "@chainsafe/lodestar-api/src/routes/validator";
+
+type DoppelgangerState = {
+  epochRegistered: Epoch;
+  epochChecked: Epoch[];
+  remainingEpochsToCheck: number;
+};
+
+export enum DoppelgangerStatus {
+  VerifiedSafe = "VerifiedSafe",
+  Unverified = "Unverified",
+  Unknown = "Unknown",
+}
+
+export class DoppelgangerService {
+  private readonly doppelgangerStateByIndex = new Map<ValidatorIndex, DoppelgangerState>();
+  constructor(
```

Comment:
Some metrics we can add:
+ Statuses of DoppelGanger for each validator
+ Time to run DoppelGanger service for each epoch

that's not necessarily to be included in this PR, just something to think about

Reply:
i think it's critical to be added to this PR, to prove that the feature is working correctly before merging